### ISSUE TITLE
Question about the historic "create" flag

### ISSUE THREAD

**Issue by philicious** [terraform-aws-modules/terraform-aws-eks#1862]
Historically there were good reasons for having the `create_cluster` flag. I know at least of the problems around the kubernetes provider when deleting a cluster, where setting `create_cluster = false` helped alot.

However, now that the kubernetes provider is gone with v18.x, I wonder if there still is a good for keeping this flag. At least I wonder if its worth keeping it in my wrapper module or if its safe to assume that I will hardly ever need in the lifecycle management of EKS and this upstream module.

In other words: Have you witnessed any notable use-case for the flag since v18 ?

**Comment by bryantbiggs** [comment]
I'm not sure I follow - in terms of conditional creation, this is just a pattern that we use throughout the modules we create/manage so its more of a "convention" than anything that gives users various levels of flexibility

**Comment by philicious** [comment]
@bryantbiggs in module version up to v17, the kubernetes provider was used, which lead to chicken&egg problems when removing / destroying a cluster from state. The create_cluster flag was quite useful to destroy all resources safely and then remove the module from code. 

Knowing the v18 code, my gut-feeling would say that this kind of problems are unlikely to happen. 

But as others have tested v18 more extensively during development, I wonder if there are any note-worthy caveats that would make the create_cluster flag useful in a similar fashion?

(Iam not challenging it's existence in general)

**Comment by bryantbiggs** [comment]
I don't think so - v18 simplified a lot of various bits down to just plain AWS Terraform resources and that lifecycle of creation/deletion is pretty well established now
